src/QsCompiler/DataStructures/Serialization.fs

```diff
@@ -26,6 +27,40 @@ type NonNullableConverter<'T when 'T: equality and 'T: null>()  =
         serializer.Serialize(writer, value.Value)
 
 
+type PositionConverter() =
+    inherit JsonConverter<Position>()
+
+    override this.ReadJson(reader, _, _, _, serializer) =
+        serializer.Deserialize<int * int> reader ||> Position.Create
+
+    override this.WriteJson(writer : JsonWriter, position : Position, serializer : JsonSerializer) =
+        serializer.Serialize(writer, (position.Line, position.Column))
+
+
+[<CLIMutable>]
```

Comment:
Makes the record into a class with mutable properties and a default zero-parameter constructor, I think. It seemed like it was necessary to get the deserialization to work (but maybe I had something else configured wrong).

Reply:
Given that the syntax tree contains plenty of records, that just seems a bit odd.

Reply:
It seems that the type being private is the cause. This type works:

```
type RangePosition =
    { Line : int
      Column : int }
```

but this type fails to deserialize:

```
type private RangePosition =
    { Line : int
      Column : int }
```

This type only exists to serialize/deserialize positions using `Line` and `Column` fields, rather than `Item1` and `Item2`, which I think is what happens when you serialize the new `Position` type directly because it is backed by a tuple. I'm not sure why Newtonsoft.Json is getting confused by a private type.

Which option do you prefer - making it public, using `CLIMutable`, or find some other way to change the serialization field names?

Reply:
That makes sense. Thanks for the explanation! CLIMutable is fine - I was just curious to understand why it was needed.